lib/ansible/modules/crypto/openssh_keypair.py

```diff
@@ -0,0 +1,355 @@
+        required: true
+        description:
+            - Name of the file in which the generated SSH private and public key will be written.
+    passphrase:
+        required: false
+        description:
+            - The passphrase for the private key.
+    comment:
+        required: false
+        description:
+            - Provides a new comment to the public key
+
+extends_documentation_fragment: files
+'''
+
+EXAMPLES = '''
+# Generate an OpenSSH keypair with the default values (4096 bits, rsa)
+- openssh_keypair:
+    path: /tmp/id_ssh_rsa
+
+# Generate an OpenSSH keypair with the default values (4096 bits, rsa)
+# and a passphrase
+- openssh_keypair:
+    path: /tmp/id_ssh_rsa
+    passphrase: ansible
+
+# Generate an OpenSSH rsa keypair with a different size (2048 bits)
+- openssh_keypair:
+    path: /tmp/id_ssh_rsa
+    size: 2048
+
+# Force regenerate an OpenSSH keypair if it already exists
+- openssh_keypair:
+    path: /tmp/id_ssh_rsa
+    force: True
+
+# Generate an OpenSSH keypair with a different algorithm (dsa)
+- openssh_keypair:
+    path: /tmp/id_ssh_dsa
+    type: dsa
+'''
+
+RETURN = '''
+size:
+    description: Size (in bits) of the SSH private key
+    returned: changed or success
+    type: int
+    sample: 4096
+type:
+    description: Algorithm used to generate the SSH private key
+    returned: changed or success
+    type: string
+    sample: rsa
+filename:
+    description: Path to the generated SSH private key file
+    returned: changed or success
+    type: string
+    sample: /tmp/id_ssh_rsa
+fingerprint:
+    description: The fingerprint of the key.
+    returned: changed or success
+    type: string
+    sample: 4096 SHA256:r4YCZxihVjedH2OlfjVGI6Y5xAYtdCwk8VxKyzVyYfM example@example.com (RSA)
+'''
+
+import os
+import sys
+import io
+import stat
+import errno
+import subprocess
+import tempfile
+import codecs
+
+from subprocess import PIPE, Popen
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from pipes import quote
+
+
+class KeypairError(Exception):
+    pass
+
+
+class Keypair(object):
+
+    def __init__(self, module):
+        self.path = module.params['path']
+        self.state = module.params['state']
+        self.force = module.params['force']
+        self.size = module.params['size']
+        self.type = module.params['type']
+        self.passphrase = module.params['passphrase']
+        self.comment = module.params['comment']
+        self.changed = False
+        self.check_mode = module.check_mode
+        self.privatekey = None
+        self.fingerprint = {}
+
+        if self.type in ('rsa', 'rsa1'):
+            self.size = 4096 if self.size is None else self.size
+            if self.size < 1024:
+                module.fail_json(msg='''For RSA keys, the minimum size is 1024 bits and the default is 4096 bits.
+                                     Attempting to use bit lengths under 1024 will cause the module to fail.''')
```

Comment:
When this message is printed, it will contain a newline followed by a lot of spaces. You should compose it as follows:
``` .py
                module.fail_json(msg=('For RSA keys, the minimum size is 1024 bits and the default is 4096 bits. '
                                      'Attempting to use bit lengths under 1024 will cause the module to fail.'))
```